share/completions/terraform.fish

```diff
@@ -1,178 +1,205 @@
-complete -f -c terraform -n '__fish_seen_subcommand_from untaint' -o lock-timeout -d 'Duration to retry a state lock'
-complete -f -c terraform -n '__fish_seen_subcommand_from untaint' -o module -d 'The module path where the resource lives'
-complete -f -c terraform -n '__fish_seen_subcommand_from untaint' -o no-color -d 'If specified, output won\'t contain any color'
-complete -f -c terraform -n '__fish_seen_subcommand_from untaint' -o state -d 'Path to a Terraform state file'
-complete -f -c terraform -n '__fish_seen_subcommand_from untaint' -o state-out -d 'Path to write state'
+complete -f -c terraform -n __fish_terraform_needs_command -a untaint -d "Manually unmark a resource as tainted"
+complete -f -c terraform -n "__fish_seen_subcommand_from untaint" -o allow-missing -d "Succeed even if resource is missing"
+complete -f -c terraform -n "__fish_seen_subcommand_from untaint" -o backup -d "Path to backup the existing state file"
+complete -f -c terraform -n "__fish_seen_subcommand_from untaint" -o lock=false -d "Don't hold a state lock"
+complete -f -c terraform -n "__fish_seen_subcommand_from untaint" -o lock-timeout -d "Duration to retry a state lock"
+complete -f -c terraform -n "__fish_seen_subcommand_from untaint" -o ignore-remote-version -d "Ignore local and remote backend compatibility check"
+complete -f -c terraform -n "__fish_seen_subcommand_from untaint" -o state -d "Path to a Terraform state file"
+complete -f -c terraform -n "__fish_seen_subcommand_from untaint" -o state-out -d "Path to write state"
 
 ### validate
-complete -f -c terraform -n __fish_use_subcommand -a validate -d 'Validate the Terraform files'
-complete -f -c terraform -n '__fish_seen_subcommand_from validate' -o no-color -d 'If specified, output won\'t contain any color'
+complete -f -c terraform -n __fish_terraform_needs_command -a validate -d "Validate the Terraform files"
+complete -f -c terraform -n "__fish_seen_subcommand_from validate" -o json -d "Produce output in JSON format"
+complete -f -c terraform -n "__fish_seen_subcommand_from validate" -o no-color -d "If specified, output won't contain any color"
 
 ### version
-complete -f -c terraform -n __fish_use_subcommand -a version -d 'Print the Terraform version'
+complete -f -c terraform -n __fish_terraform_needs_command -a version -d "Print the Terraform version"
+
+### workspace
+complete -f -c terraform -n __fish_terraform_needs_command -a workspace -d "Workspace management"
+complete -f -c terraform -n "__fish_seen_subcommand_from workspace" -a list -d "List workspaces"
+complete -f -c terraform -n "__fish_seen_subcommand_from workspace" -a select -d "Select an workspace"
+complete -f -c terraform -n "__fish_seen_subcommand_from workspace" -a new -d "Create a new workspace"
+complete -f -c terraform -n "__fish_seen_subcommand_from workspace" -a delete -d "Delete an existing workspace"
```

Comment:
Completely untested but I imagine something like this could be useful

	complete -f -c terraform -n "__fish_seen_subcommand_from workspace && __fish_seen_subcommand_from select delete" -a "(terraform workspace list)"

Reply:
I really like the idea. It needed a bit more work, but I've now added an autocomplete for the workspace select and delete. We may be able to do something similar with state, but it's a bit less straightforward.